src/libs/actions/Transaction.ts

```diff
@@ -245,14 +246,28 @@ function getOnyxDataForRouteRequest(transactionID: string, isDraft = false): Ony
     };
 }
 
+/**
+ * Sanitizes the waypoints by removing the pendingAction property.
+ *
+ * @param waypoints - The collection of waypoints to sanitize.
+ * @returns The sanitized collection of waypoints.
+ */
+function sanitizeRecentWaypoints(waypoints: WaypointCollection): WaypointCollection {
+    return Object.entries(waypoints).reduce((acc, [key, waypoint]) => {
+        const {pendingAction, ...rest} = waypoint as RecentWaypoint;
+        acc[key] = rest;
+        return acc;
+    }, {} as WaypointCollection);
```

Comment:
There was no need for casting here, the reducer function just needed the type declared:

`(acc: WaypointCollection, [key, waypoint]) => {...}`

then the cast is not needed